crates/bdk/src/wallet/mod.rs

```diff
@@ -126,12 +126,18 @@ pub struct ChangeSet {
     /// [`IndexedTxGraph`]: bdk_chain::indexed_tx_graph::IndexedTxGraph
     pub indexed_tx_graph:
         indexed_tx_graph::ChangeSet<ConfirmationTimeAnchor, keychain::ChangeSet<KeychainKind>>,
+
+    /// Stores the network type of the wallet.
+    pub network: Option<Network>,
 }
 
 impl Append for ChangeSet {
     fn append(&mut self, other: Self) {
         Append::append(&mut self.chain, other.chain);
         Append::append(&mut self.indexed_tx_graph, other.indexed_tx_graph);
+        if other.network.is_some() {
```

Comment:
Should probably `debug_assert` at least that these are not conflicting.

A cleaner less brittle approach would be to have some fixed `Header` data that you can set in persistence. Make an issue if you think it's worth discussing.